ranger-core/src/main/java/io/appform/ranger/core/finder/serviceregistry/ServiceRegistryUpdater.java

```diff
@@ -132,16 +132,17 @@ private void updateRegistry() throws InterruptedException {
                         serviceRegistry.getService().getServiceName());
             return;
         }
-        val nodeList = nodeDataSource.refresh(deserializer);
+        val nodeList = nodeDataSource.refresh(deserializer).orElse(null);
```

Comment:
I tried optional.ifPresent() but the log had to be again captured inside (!optional.isPresent()) because this is java8, some of the niceties of later java versions to capture this flow elegantly are not present. So left it like this. Do advise if you want this changed.